### PR TITLE
Create Android machine_uid.md

### LINKED ISSUE #844: KDD: Android machine uid
This is already implemented, it'd just be good to get a KDD summarising the contending solutions that were considered and why we've settled on the ANDROID_ID approach for the time being. So that we can remember what we've done easy!

related:
- https://github.com/openmsupply/open-msupply/issues/131
- https://github.com/openmsupply/open-msupply/issues/221

### PR DESCRIPTION
**Pull request by Chris-Petty**
fixes #844

### CODE REVIEW

**Review by mark-prins — APPROVED** [review]
👍  LGTM! thanks @Chris-Petty

File: decisions/Android machine_uid.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+_Pros:_
+
+- One very simple implementation for all platforms
+
+_Cons:_
+
+- Changes on factory resets
+- Changes on clearing app data/reinstalling app
+- Breaks the protection of stopping sync when a database is shared (in desktop deployments with postgres/sqlite)
```
Comment (andreievg) [inline decisions/Android machine_uid.md:65]: We actually did an implementation of this at the start, but it got voted out. 
I am not sure what this con mean, btw in that implementation app data was a file, maybe that helps clarify this Option implementation ?

**Review by andreievg — APPROVED** [review]
LGTM, sounds like Option 1 still holding strong !
Just FYI it was actually quite useful to know hardware id in Native Java layer, as it can be used to verify that server is local

File: decisions/Android machine_uid.md (hunk 1 above)
Comment (Chris-Petty) [reply decisions/Android machine_uid.md:65]: Yea app data in a file would fix this last point, unless someone unwittingly moves that file too

**Pull request merged**